test-cases/tests/alerts/c18-validate-api-usage-alerts.md

```diff
@@ -0,0 +1,126 @@
+---
+products:
+  - name: rhoam
+    environments:
+      - osd-fresh-install
+estimate: 1h
+tags:
+  - per-release
+---
+
+# C18 - Validate API usage alerts
+
+## Prerequisites
+
+1. Login via `oc` as a user with **cluster-admin** role (kubeadmin):
+
+```
+oc login --token=<TOKEN> --server=https://api.<CLUSTER_NAME>.s1.devshift.org:6443
+```
+
+2. RHOAM is installed using the [Jenkins pipeline](https://master-jenkins-csb-intly.cloud.paas.psi.redhat.com/job/ManagedAPI/job/managed-api-install-master/build?delay=0sec) with following parameters:
+
+- integreatlyOperatorBranchName: (Go to integreatly-operator github repo and select latest RHOAM release branch)
+- ocmClusterLifespan: 36
+- openshiftVersion: (search for the version in the test plan document)
+- useByoc: false
+- multiAZ: false
+- patchCloudResAwsStrCM: false
+- clusterID: api-usage-alerts
+- emailRecipients: <your@email.address>
+- stepsToDo: provision + install
+
+## Steps
+
+1. Validate that rate-limiting alerts ConfigMap has been created
+
+```
+oc get cm rate-limit-alerts -n redhat-managed-api-operator
+```
+
+2. Verify that level1, level2 and level3 rate-limiting alerts are present
+
+```
+oc get prometheusrules -n redhat-managed-api-marin3r
+```
+
+3. Modify the `sku-limits-managed-api-service` configmap to set the rate limit to 20 requests per minute.
+   Run
+
+```
+oc edit configmap sku-limits-managed-api-service -n redhat-rhmi-operator
+```
+
+And insert the following data:
+
+```
+data:
+  rate_limit: |-
+    {
+      "RHOAM SERVICE SKU": {
+        "unit": "minute",
+        "requests_per_unit": 20
+      }
+    }
+```
+
+4. Modify the `rate-limit-alerts` to allow alerts to fire in a reasonable time from the testing perspective:
+
+```
+oc edit configmap rate-limit-alerts -n redhat-rhmi-operator
+```
+
+and insert the following data:
+
+```
+{
+      "api-usage-alert-level1": {
+        "ruleName": "Level1ThreeScaleApiUsageThresholdExceeded",
+        "level": "warning",
+        "minRate": "80%",
+        "maxRate": "90%",
+        "period": "1m"
+      },
+      "api-usage-alert-level2": {
+        "ruleName": "Level2ThreeScaleApiUsageThresholdExceeded",
+        "level": "warning",
+        "minRate": "90%",
+        "maxRate": "95%",
+        "period": "2h"
```

Comment:
```suggestion
        "period": "1m"
```

Reply:
I'd set the period for the other levels to 1m beforehand as they're going to be tested as well